apps/backend/src/db/migrations/20250606_init.ts

```diff
@@ -0,0 +1,304 @@
+import { type CreateTableBuilder, Kysely, sql } from 'kysely';
+
+const addDefaultColumns = (ctb: CreateTableBuilder<any, any>) => {
+  return ctb
+    .addColumn('id', 'uuid', (col) => col.primaryKey().defaultTo(sql`gen_random_uuid()`))
+    .addColumn('createdAt', 'date', (col) => col.notNull().defaultTo(sql`CURRENT_TIMESTAMP`))
+    .addColumn('modifiedAt', 'date', (col) => col.notNull().defaultTo(sql`CURRENT_TIMESTAMP`)); // I don't know how to set this to update automatically
+};
+
+const addNameCheckConstraint = (ctb: CreateTableBuilder<any, any>) => {
+  return ctb.addCheckConstraint('valid_name', sql`"name" ~ '^[a-zA-Z0-9 .,_:;!?()/\-]{1,256}$'`);
+};
+
+// --- Table Creation Helper Functions ---
+async function createUserTable(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .createTable('User')
```

Comment:
I would introduce an enum here to be able to reuse the table name later on more easily. We may even do this with the columns, as they will be reused within the definition of the FKs. This would also help with refactoring the migration later on.